front/src/reducers/osrdconf/helpers.ts

```diff
@@ -113,3 +114,46 @@ export const insertViaFromMap = (
 
   return addElementAtIndex(pathSteps, newViaIndex, newStep);
 };
+
+export const updatePathStepAtIndex = (
+  state: OsrdConfState,
+  index: number,
+  updates: Partial<PathStep> | null,
+  replaceCompletely: boolean = false
+) => {
+  if (replaceCompletely) {
+    state.pathSteps = replaceElementAtIndex(state.pathSteps, index, updates as PathStep);
+  } else {
+    const element = state.pathSteps[index];
+    if (element) {
+      const updatedElement = { ...element, ...updates };
+      state.pathSteps = replaceElementAtIndex(state.pathSteps, index, updatedElement);
+    }
+  }
+};
+
+export const updateOriginPathStep = (
+  state: OsrdConfState,
+  origin: Partial<PathStep> | null,
+  replaceCompletely: boolean = false
+) => {
+  updatePathStepAtIndex(state, 0, origin, replaceCompletely);
+};
+
+export const updateDestinationPathStep = (
+  state: OsrdConfState,
+  destination: Partial<PathStep> | null,
+  replaceCompletely: boolean = false
+) => {
+  updatePathStepAtIndex(state, state.pathSteps.length - 1, destination, replaceCompletely);
+};
+
+export const getPathStepArrivalData = (arrivalTime: PathStep['arrival']) => ({
```

Comment:
I'm not sure we should do that :thinking: 
I think that if the arrivalTime is modified, we should only update it